### PR TITLE
#[187364723] - Setup docker for backend application

### PR DESCRIPTION
**Pull request by AngeloChristian1**
**What does this PR do** 
Setting Up docker configurations 

**Task Included**
1. Initializing a Dockerfile
2. Setting up a docker-compose.yaml file that runs two services
    - Express Server
    - Postgress Database
  
**How should this be manually tested?**
- Before cloning the repo make sure that you have installed docker in your local machine
- Make sure that you fill in all necessary environment variables required
- Clone the repo and install the required packages
- cd into the repo in the terminal and run 
       - 'docker-compose up --build' : which will build the images and container
       - 'docker run -p e-commerce-bitcrafters-bn' : which will run the image
- Other necessary commands
    - 'docker login' : which will allow you to log into your docker account (Better if you have a desktop docker app)
    - 'docker compose up': It wakes up the container
    - 'docker compose down': It sleeps the container

**Any background context you want to provide?**
- Install docker
- Add docker extension on your editor for easy configurations

**Pivotal Tracker stories**
#187364723

### CODE REVIEW

**Review by mpanoathos — APPROVED** [review]

**Review by Yvan-David — APPROVED** [review]

File: .structure.env
Comment (munezerobagira) [inline .structure.env:1]: its common to call this .env.example

File: .circleci/config.yaml
Comment (munezerobagira) [inline .circleci/config.yaml:1]: why do we have new file? seem to duplicate of config.yml

File: package.json [hunk 1]
```diff
@@ -4,9 +4,10 @@
   "description": "",
   "main": "dist/index.js",
   "scripts": {
-    "start": "node dist/index.js",
-    "dev": "nodemon --exec ts-node index.ts",
-    "build": "tsc",
+    "start": "tsc && node dist/index.js",
+    "dev": "nodemon --exec ts-node index.ts --watch src --ext ts,js",
+    "build":"tsc",
+    "babel-build": "rm -rf build/ && babel src --out-dir build/src --extensions '.ts' --copy-files --source-maps && babel index.ts --out-file build/index.js --extensions '.ts' --copy-files",
```
Comment (munezerobagira) [inline package.json]: why do we have 2 build, babel-build and build?

File: src/config/database.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  dialect: 'postgres',
+  host: 'database' || 'localhost',
+  port: 5432,
+  username: POSTGRES_USERNAME,
+  password: POSTGRES_PASSWORD,
+  database: POSTGRES_DATABASE,
+
+});
+
+export default  sequelize ;
```
Comment (munezerobagira) [inline src/config/database.ts]: why do we have this new file?

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+node_modules
+./node_modules
```
Comment (munezerobagira) [inline .dockerignore:2]: do we need to have all these, isn't a duplicate?

File: .gitignore [hunk 4]
```diff
@@ -12,3 +12,9 @@ coverage
 .env
 /.env
 ./.env
+
+build
+./build
+
+dist
+./dist
```
Comment (munezerobagira) [inline .gitignore]: can you check if this is not duplicate?

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**